pkg/query/selection.go

```diff
@@ -175,13 +187,26 @@ func (c Criterion) Validate() error {
 		return &util.UnsupportedQueryError{Message: fmt.Sprintf("separator %s is not allowed in %s with left operand \"%s\".", Separator, c.Type, c.LeftOp)}
 	}
 	for _, op := range c.RightOp {
-		if strings.ContainsRune(op, '\n') {
+		if strings.ContainsRune(op, '\n') && c.Type != ExistQuery {
 			return &util.UnsupportedQueryError{Message: fmt.Sprintf("%s with key \"%s\" has value \"%s\" contaning forbidden new line character", c.Type, c.LeftOp, op)}
 		}
 	}
 	return nil
 }
 
+func (c *Criterion) GetTemplateParams() map[string]interface{} {
+	return c.TemplateParametersMap
+}
+
+func (c *Criterion) AddTemplateParams(templateParametersMap map[string]interface{}) {
```

Comment:
same here, is this needed?

Reply:
I've changed it to be Private (as now it's being used as part of ByIdNotExistWithTemplateParameters and ByIdExistWithTemplateParameters)